src/main/java/util/TickingTimer.java

```diff
@@ -158,4 +158,14 @@ public boolean isStarted() {
         return started;
     }
 
+    /**
+     * This function is used to dynamically change the period based on the api quota.
+     * Triggering this function will restart the timer based on the new period.
+     * @param period : the amount of time before triggering the timer.
+     */
```

Comment:
Method's header is preferable not about how it's currently used but what it does e.g. can start with `Changes the timer's period dynamically...`. Also shouldn't mention the `api quota` since this method has no knowledge of the api quota.